File: server/DocumentsKM/Models/ConstructionSubtype.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DocumentsKM.Models
{
    // Подвид конструкции
    public class ConstructionSubtype
    {
        [Key]
        public Int16 Id { get; set; }

        // Вид конструкции
        [Required]
        [ForeignKey("TypeId")]
        public virtual ConstructionType Type { get; set; }

        // Название
        [Required]
        [MaxLength(255)]
        public string Name { get; set; }

        // Расценка
        [Required]
        [MaxLength(10)]
        public string Valuation { get; set; }
    }
}

File: client/src/model/ConstructionSubtype.ts
interface IConstructionSubtype {
	id: number
	name: string
	valuation: string
}

class ConstructionSubtype {
	id: number
	name: string
	valuation: string

	constructor(obj?: IConstructionSubtype) {
		this.id = (obj && obj.id) || 0
		this.name = (obj && obj.name) || ''
		this.valuation = (obj && obj.valuation) || ''
	}
}

export default ConstructionSubtype
